modules/puppetca/puppetca_api.rb

```diff
@@ -15,34 +16,12 @@ class Api < ::Sinatra::Base
       end
     end
 
-    get "/autosign" do
-      content_type :json
+    post "/autosign" do
+      params = JSON.parse(request.env["rack.input"].read)
```

Comment:
Since the api is being changed, why not change it to something like:

```post "/autosign/:csr"``` instead of JSON encoding and then decoding a single parameter.

Reply:
I'm pretty sure that CSRs are not URL-safe, as they include slashes and line-breaks.

Reply:
You could UR-encode the parameter? Alternatively, instead of putting the parameter in the url, you could put it in the body of the request (would still need to be url-encoded though), which is a bit cleaner.